src/libmongoc/src/mongoc/mongoc-trace-private.h

```diff
@@ -89,6 +89,23 @@ BSON_BEGIN_DECLS
                   (int) _l);                              \
       mongoc_log_trace_bytes (MONGOC_LOG_DOMAIN, _b, _l); \
    } while (0)
+#define DUMP_BSON(_bson)                                            \
+   do {                                                             \
+      char *_bson_str;                                              \
+      if (_bson) {                                                  \
+         _bson_str = bson_as_canonical_extended_json (_bson, NULL); \
+      } else {                                                      \
+         _bson_str = bson_strdup ("<NULL>");                        \
+      }                                                             \
+      mongoc_log (MONGOC_LOG_LEVEL_TRACE,                           \
+                  MONGOC_LOG_DOMAIN,                                \
+                  "TRACE: %s():%d %s = %s",                         \
+                  BSON_FUNC,                                        \
+                  __LINE__,                                         \
+                  #_bson,                                           \
+                  _bson_str);                                       \
+      bson_free (_bson_str);                                        \
+   } while (0)
```

Comment:
Trace logs have come in handy for debugging some monitoring bugs. It seemed worthwhile to add to them for future debugging.